pkg/api/proxy.go

```diff
@@ -13,6 +11,7 @@ import (
 	"zotregistry.dev/zot/pkg/api/constants"
 	"zotregistry.dev/zot/pkg/cluster"
 	"zotregistry.dev/zot/pkg/common"
+	"zotregistry.dev/zot/pkg/proxy"
```

Comment:
What is this split between pkg/api/proxy.go and pkg/proxy/proxy.go?

Reply:
During development, there was a circular import for api and the new gql proxy in the extensions. This led me to break up the generic proxy logic into its own package and call it from the api package as well as from the gql proxy package.

That said, I do agree that the file naming could potentially be better.